addons/l10n_in_edi_ewaybill/models/account_edi_format.py

```diff
@@ -0,0 +1,601 @@
+                "message": _("waiting For IRN generation To create E-waybill")}
+            ]}
+        token = self._l10n_in_edi_get_token(company)
+        if not token:
+            return self._l10n_in_edi_no_config_response()
+        params = {
+            "auth_token": token,
+            "json_payload": json_payload,
+        }
+        return self._l10n_in_edi_connect_to_server(company, url_path="/iap/l10n_in_edi/1/generate_ewaybill_by_irn", params=params)
+
+    @api.model
+    def _l10n_in_edi_irn_ewaybill_get(self, company, irn):
+        token = self._l10n_in_edi_get_token(company)
+        if not token:
+            return self._l10n_in_edi_no_config_response()
+        params = {
+            "auth_token": token,
+            "irn": irn,
+        }
+        return self._l10n_in_edi_connect_to_server(company, url_path="/iap/l10n_in_edi/1/get_ewaybill_by_irn", params=params)
+
+    #=============================== E-waybill API methods ===================================
+
+    @api.model
+    def _l10n_in_edi_ewaybill_no_config_response(self):
+        return {"error": [{
+            "code": "000",
+            "message": _(
+                "A username and password still needs to be set or it's wrong for the E-waybill(IN). "
+                "It needs to be added and verify in the Settings."
+            )}
+        ]}
+
+    @api.model
+    def _l10n_in_edi_ewaybill_check_authentication(self, company):
+        sudo_company = company.sudo()
+        if sudo_company.l10n_in_edi_ewaybill_username and sudo_company._l10n_in_edi_ewaybill_token_is_valid():
+            return True
+        elif sudo_company.l10n_in_edi_ewaybill_username and sudo_company.l10n_in_edi_ewaybill_password:
+            authenticate_response = self._l10n_in_edi_ewaybill_authenticate(company)
+            if not authenticate_response.get("error"):
+                return True
+        return False
+
+    @api.model
+    def _l10n_in_edi_ewaybill_connect_to_server(self, company, url_path, params):
+        user_token = self.env["iap.account"].get("l10n_in_edi")
+        params.update({
+            "account_token": user_token.account_token,
+            "dbuuid": self.env["ir.config_parameter"].sudo().get_param("database.uuid"),
+            "username": company.sudo().l10n_in_edi_ewaybill_username,
+            "gstin": company.vat,
+        })
+        if company.sudo().l10n_in_edi_production_env:
+            default_endpoint = DEFAULT_IAP_ENDPOINT
+        else:
+            default_endpoint = DEFAULT_IAP_TEST_ENDPOINT
+        endpoint = self.env["ir.config_parameter"].sudo().get_param("l10n_in_edi_ewaybill.endpoint", default_endpoint)
+        url = "%s%s" % (endpoint, url_path)
+        try:
+            return jsonrpc(url, params=params, timeout=25)
+        except AccessError as e:
+            _logger.warning("Connection error: %s", e.args[0])
+            return {
+                "error": [{
+                    "code": "404",
```

Comment:
Is not this 404 misleading here, it can be any other http code...

Reply:
404 is mins that the server is offline or not found so how it's misleading!?

Reply:
Because 404 in html means page not found and it is not at all certain that this is the error reported by the AccessError. If the 404 is done by us, why not just put access_error like no_credit we already have.

Reply:
okay